ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -165,6 +165,14 @@ impl InstrIdx {
     pub(crate) fn instr<'a>(&'a self, jit_mod: &'a Module) -> &Instruction {
         jit_mod.instr(*self)
     }
+
+    /// Make a local variable [Operand] from a local variable index.
+    pub(crate) fn to_operand(&self, m: &Module) -> Operand {
+        // If the instruction doesn't define a local variable, then it doesn't make sense to make
+        // an operand of it.
+        debug_assert!(self.instr(m).is_def());
```

Comment:
IMHO this should be `#Panics` or "undefined behaviour" doc string too, as if you call this function incorrectly you will get a surprise I think?

Reply:
It won't cause UB, it just means that you can pass an operand of void type, which doesn't make much sense. When would you ever need to do that?

I considered always checking and panicking, but `self.instr(m).is_def()` comes at a small runtime cost.

Reply:
My point is more: I think functions must document (even briefly) their preconditions and (when possible) error in some sensible way if those are violated. Otherwise what happens is our future selves forget the preconditions, call the function in the wrong way, and then spend days debugging because they got the wrong end of the stick.

Reply:
I'm not sure it's a precondition. Maybe we should just delete the assertion.

@ptersilie what do you think?

Reply:
Hm, I think `debug_assert` is fine here. You can make an operand from a non-def instruction, it just doesn't make sense. Do we ever need the instridx of a non-def instruction, or can we just make sure we never get the instruction index of a non-def instruction?

Reply:
If we don't know we need to do something, I'd rather error if someone does it. If they then work out that they can generalise the function, they'll be able to do so knowing the full context.

Reply:
How would we prefer to do this?
 - change `debug_assert` to `assert` and take some runtime hit.
 - keep it `debug_assert` and document that it's UB to make a void operand (a slight untruth).

Reply:
I've said my peace so please do whatever you think best (inc "keep as-is").

Reply:
Since I've decided to revert this commit, this function too disappears, and this conversation is resolved.

Revert in 6409770a974abce2a137b21c190a99069f73c8ce